client/src/components/app/paginated-task-list/paginated-task-list.jsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { fetchTasks } from "../../../data/fetchTasks";
+import { TaskList } from "./task-list/task-list";
+import { Pagination } from "./pagination/pagination";
+import { PAGE_SIZE_OPTIONS } from "../../../constants/constants";
+import { useQuery } from "@tanstack/react-query";
+
+export function PaginatedTaskList() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageCount, setPageCount] = useState(1);
+  const [pageSize, setPageSize] = useState(PAGE_SIZE_OPTIONS[0]);
+  const [tasks, setTasks] = useState([]);
+
+  function handlePageChanged(pageNumber) {
+    setCurrentPage(pageNumber);
+  }
+
+  function handlePageSizeChanged(size) {
+    setPageSize(size);
+  }
+
+  const { isPending, isError, error, data } = useQuery({
+    queryKey: ["tasks", { currentPage, pageSize }],
+    queryFn: () => fetchTasks(currentPage, pageSize),
+  });
+
+  useEffect(() => {
+    if (data) {
+      if (currentPage > data.meta.pagination.pageCount) {
+        setCurrentPage(data.meta.pagination.pageCount);
+      }
+      setTasks(data.data);
+      setPageCount(data.meta.pagination.pageCount);
+    }
+  }, [data, currentPage]);
+
+  if (isPending) {
+    return <span>Loading...</span>;
+  }
+
+  if (isError) {
+    return <span>Error: {error.message}</span>;
+  }
+
+  // At this point we can assume data is not falsy
```

Comment:
You can remove this comment as it's copied from the class' implementation, but you're not using `data` below this line.